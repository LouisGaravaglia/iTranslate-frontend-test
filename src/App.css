html, body {
  min-width: 100vh;
  width: 100vw;
  height: 100vh;
  margin: 0px;
  padding: 0px !important;
  overflow-x: hidden;
}

.Browse-Albums {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  flex: 1 0 60vh;
}

.Album {
  position: relative;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  flex: 1 0 21%;
  margin: 40px;
}

.Album-Name-Box {
  position: relative;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  position: absolute;
  top:0;
  left: 0;
  height: 300px;
  width: 300px;
  background-color: #3800a7;
  opacity: 0.7;
  cursor: pointer;
}

.Album-Name {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 700;
  font-size: 2rem;
  z-index: 6;
  margin: 5px 5px;
}

.Artist-Artists {
  background: linear-gradient(to right, #ff4bde 0%, #b84dfa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Artist-Albums {
  background: linear-gradient(to right, #fdc3df 0%, #b84dfa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Artist-Tracks {
  background: linear-gradient(to right, #6a2fdc 0%, #8943b5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Artist-Language {
  background: linear-gradient(to right, #bf00ff 0%, #e167fa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Artist-Language-Text {
  color: #dba3e6;
}

.Artist-Language-Button {
  color: #dba3e6;
}

.Artist-Lyrics {
  background: linear-gradient(to right, #4f1efe 0%, #7c3bc5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Artist-Translation {
  background: linear-gradient(to right, #7c3bc5 0%, #ce5b93 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Danceability-Track-Subheader {
  background: linear-gradient(to right, #4f1efe  0%, #7c3bc5  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Danceability-Track-Header {
  background: linear-gradient(to right, #4f1efe 0%, #7c3bc5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Danceability-Language {
  background: linear-gradient(to right, #7c3bc5 0%, #d383fa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Danceability-Language-Text {
  color: #9eb5fa;
}

.Danceability-Language-Button {
  color: #9eb5fa;
}

.Danceability-Lyrics {
  background: linear-gradient(to right, #4f1efe 0%, #7c3bc5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Danceability-Translation {
  background: linear-gradient(to right, #7c3bc5 0%, #f263a8 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Browse-Categories {
  background: linear-gradient(to right, #16bdfa 0%,#357ffe 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Genres {
  background: linear-gradient(to right, #ff4bde 0%, #b84dfa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Artists {
  background: linear-gradient(to right, #ffa114 0%, #d563c0 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Albums {
  background: linear-gradient(to right, #bf6571 0%, #ffb22e 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Tracks {
  background: linear-gradient(to right, #6a2fdc 0%, #8943b5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Language {
  background: linear-gradient(to right, #9161db 0%, #c594bf 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Genre-Language-Text {
  color: #e3b2af;
}

.Genre-Language-Button {
  color: #e3b2af;
}

.Genre-Lyrics {
  background: linear-gradient(to right, #4f1efe 0%, #7c3bc5 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Genre-Translation {
  background: linear-gradient(to right, #7c3bc5 0%, #c1666f 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Browse-Landing {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  width: 100vw;
  height: 100vh;
}

.Browse-Landing p {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 900;
  font-size: 4rem;
  margin: 0px 40px;
  padding: 6px 12px;
  flex: 0 1 100%;
}

.Browse-Danceability-v2 {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  width: 100vw;
  height: 100vh;
}

.Browse-Danceability-v2 h1{
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 900;
  font-size: 3rem;
  background: linear-gradient(to right, #4f1efe  0%, #5d3bc5  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  text-align: center;
  margin: 20px 0 0;
}


.Pagination-Slider-Placeholder-v2 {
  width: 100vw;
  flex: 2;
}

.Temp-v2 {
  width: 100vw;
  flex: 2;
}

.Danceability-Pagination-Slider-Container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  width: 100vw;
  flex: 2;
}

.Danceability-Pagination-Slider {
  flex: 0 0 30vw;
}

.Danceability-Slider-Container-v2 {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  width: 100vw;
}

.Danceability-Slider-Item-v2 {
  flex: 0 0 80vw;
}

.Danceability-No-Results-Container {
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  width: 100vw;
  height: 100%;
  flex: 2;
}

.Danceability-No-Results {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 100;
  font-size: 5rem;
  background: linear-gradient(to right, #4f1efe  0%, #7c3bc5  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: default;
}

.Danceability-Result-Box {
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  width: 100vw;
  flex: 2;
}

.Danceability-Result-Container {
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
}

.Flash-Messages-Container {
  position: absolute;
  position: sticky;
  top: 0;
  z-index: 6;
}

.FlashMessage {
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: nowrap;
  background-color: #e767f0;
  height: 15vh;
  width: 100vw;
}

.FlashMessage p {
  color: white;
  margin: 0 30px;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 800;
  font-size: 2rem;
}

.FlashMessage span {
  color: white;
  margin: 0 30px;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 500;
  font-size: 2rem;
}

.Loading-Box {
  position: absolute;
  z-index: 2;
  left:50%;
  top:50%;
  transform: translate(-50%, -50%);
}

.Lyrics-Translation {
  display: flex;
  flex-wrap: nowrap;
  justify-content: space-around;
  min-width: 100vw;
  min-height: 100vh;
}

.Lyrics-Container {
  margin: 0 20px 20px 50px;
  flex: 1 1 50vw;
}

.Lyrics {
  white-space: pre-line;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 600;
  font-size: 1.5rem;
  margin: 40px 40px;
}

.Main-Container{
  position: relative;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  flex-wrap: wrap;
  width: 100vw;
  height: 100vh;
}

.Main-Result-Container {
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  flex-direction: column;
  width: 100vw;
  height: 100%;
}

.Result-Main {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 100;
  font-size: 5rem;
  color: #fff;
  flex: 1 0 200px;
  margin: 0 40px;
  padding: 0;
}

.Hover-Box {
  flex: 0 1 auto;
  margin: 0;
  padding: 0;
}

.Result-Main-TopFiller {
  flex: 10 1 auto;
}

.Main-Pagination-Slider-Container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  width: 100vw;
  flex: 3 0 20vh;
}

.Main-Pagination-Slider {
  flex: 0 0 30vw;
}

.Main-Pagination-Slider-Placeholder {
  text-align: center;
  flex: 2 0 20vh;
}

.Navbar-Container-Left {
  position: absolute;
  height: 80px;
}

.Navbar-Container-Right {
  position: absolute;
  right: 0;
  height: 80px;
}

a {
  text-decoration: none;
}

.Navbar-Search-Box {
  position: sticky;
  z-index: 4;
  top: 30px;
  margin: 0;
  padding: 0;
  margin-left: 50px;
}

.Navbar-Search {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 800;
  font-size: 1.5rem;
  color: #fff;
  margin: 0;
  padding: 0;
}

.Navbar-Browse-Box {
  position: sticky;
  z-index: 4;
  top: 30px;
  margin: 0;
  padding: 0;
  margin-right: 50px;
}

.Navbar-Browse {
  float: right;
  cursor: pointer;
  color: #fff;
}

.Result-Box {
  text-align: center;
  width: 100vw;
  flex: 4;
}

.Result-Header {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 100;
  font-size: 5rem;
  margin: 40px 0;
  padding: 0;
}

.Result-Subheader {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 200;
  font-size: 2rem;
  cursor: default;
  margin: 0 40px;
  padding: 0;
}

.Search-Field {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  flex-direction: column;
  width: 100vw;
  height: 100vh;
}

.Search-Field-Content h1 {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 800;
  font-size: 5rem;
  display: block;
  text-align: center;
  cursor: default;
 }

.Search-Field-Content {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  flex-direction: column;
  flex: 1;
  margin: 0 80px;
}

.Search-Field input {
  width: 30vw;
  height: 50px;
  border: none;
}

.Search-Field button{
  background-color: #fff;
  border: none;
  font-size: 1rem;
  padding: 6px 12px;
  margin: auto;
  position: absolute;
  text-align: center;
  right:0;
  top:50%;
  transform: translate(-10%, -50%);
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  cursor: pointer;
}

.Search-Field-Filler {
  flex: 0.1 0 1vh;
}

#SearchVal {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 400;
  font-size: 1.3rem;
  padding-left: 25px;
  outline:none;
  padding-right: 60px;
}

.icon {
  font-size: 1.5rem;
}

.Search-Input-Container {
  position: relative;
}

.Search-Landing {
  background: linear-gradient(to right, #ff76e4 0%, #d437ff 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Search-Landing-Text {
  color: #A800FF;
}

.Search-Landing-Button {
  color: #A800FF;
}

.Search-Track-Subheader {
  background: linear-gradient(to right, #ff7ebc 0%, #ee67fa  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Search-Track-Header {
  background: linear-gradient(to right, #ff7ebc  0%, #ee67fa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  cursor: pointer;
}

.Search-Language {
  background: linear-gradient(to right, #ee67fa 0%, #bf00ff 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Search-Language-Text {
  color: #6000FF;
}

.Search-Language-Button {
  color: #6000FF;
}

.Search-Lyrics {
  background: linear-gradient(to right, #d1f1fc 0%, #80d9ff  100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.Search-Translation {
  background: linear-gradient(to right, #80d9ff  0%, #f04fff 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.TopArrow-Container {
  position: relative;
  height: 150px;
  width: 100vw;
}

.TopArrow-Box {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
  position: fixed;
  left:50%;
  transform: translate(-50%, -50%);
  bottom: 0;
  z-index: 4;
  text-decoration: none;
  width: 80px;
  height: 60px;
  cursor: pointer;
}

.Translation-Container {
  margin: 0 50px 20px 20px;
  flex: 1 1 50vw;
}

.Translation {
  white-space: pre-line;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: 600;
  font-size: 1.5rem;
  margin: 40px 40px;
}

.inViewPlaceholder {
  height: 10px;
  width: 100vw;
}

/* //////////////////////////////////////  MAX WIDTH 768PX  ////////////////////////////////////// */

@media only screen and (max-width: 768px) {

  .Search-Field-Content h1 {
    font-size: 4rem;
  }

  .Result-Header {
    font-size: 4rem;
  }

  .Result-Subheader {
    font-size: 1.5rem;
  }

  .Result-Main {
    font-size: 4rem;
  }

  .Danceability-No-Results {
    font-size: 4rem;
  }

  .Translation-Container {
    margin: 0 10px 20px 5px;
  }

  .Lyrics-Container {
    margin: 0 10px 20px 5px;
  }

  .Lyrics {
    font-size: 1rem;
    margin: 20px 20px;
  }

  .Translation {
    font-size: 1rem;
    margin: 20px 20px;
  }

  .FlashMessage p {
    font-size: 1.5rem;
  }

}

/* //////////////////////////////////////  MAX WIDTH 623PX  ////////////////////////////////////// */

@media only screen and (max-width: 623px) {

  .Search-Field-Content h1 {
    font-size: 3rem;
  }

  .Result-Header {
    font-size: 3rem;
  }

  .Result-Subheader {
    font-size: 1rem;
  }

  .Result-Main {
    font-size: 3rem;
  }

  .Danceability-No-Results {
    font-size: 3rem;
  }

  .Browse-Landing p {
    font-size: 3rem;
  }

  .Lyrics-Translation {
    flex-wrap: wrap;
    justify-content: flex-start;
  }

  .Translation-Container {
    flex: 1 0 100%;
    margin: 0 10px 20px 5px;
    max-width: 450px;
  }

  .Lyrics-Container {
    flex: 1 0 100%;
    margin: 0 10px 20px 5px;
    max-width: 450px;
  }

  .Lyrics {
    font-size: 1rem;
    margin: 20px 20px;
  }

  .Translation {
    font-size: 1rem;
    margin: 20px 20px;
  }

  .FlashMessage p {
    font-size: 1rem;
  }

}

/* //////////////////////////////////////  MAX WIDTH 500PX  ////////////////////////////////////// */

@media only screen and (max-width: 500px) {

  .Search-Field-Content h1 {
    font-size: 2.5rem;
  }

  .Browse-Landing p {
    font-size: 2.5rem;
  }

  .Translation-Container {
    flex: 1 0 100%;
    margin: 0 10px 20px 5px;
    max-width: 300px;
  }

  .Lyrics-Container {
    flex: 1 0 100%;
    margin: 0 10px 20px 5px;
    max-width: 300px;
  }

  .Lyrics {
    font-size: 1rem;
    margin: 20px 20px;
  }

  .Translation {
    font-size: 1rem;
    margin: 20px 20px;
  }

  .FlashMessage p {
    font-size: 1rem;
  }

}