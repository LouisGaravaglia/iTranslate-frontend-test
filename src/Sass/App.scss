@import "components/navbar";
@import "components/toTopArrow";
@import "components/browseLanding";
@import "components/lyrics";
@import "components/translation";
@import "components/searchBar";
@import "components/flashMessage";
@import "components/danceabilityRoute";
@import "components/result";
@import "components/mainContainer";
@import "components/album";
@import "components/loadingIcon";
@import "components/searchFontColors";
@import "components/browseByDanceabilityFontColors";
@import "components/browseByArtistsFontColors";
@import "components/browseByGenreFontColors";
@import "mediaQueries";

html, body {
  min-width: 100vh;
  width: 100vw;
  height: 100vh;
  margin: 0px;
  padding: 0px !important;
  overflow-x: hidden; 
}