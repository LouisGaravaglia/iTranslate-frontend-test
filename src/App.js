import React, {useEffect} from 'react';
import './App.css';
//COMPONENT IMPORTS
import Routes from "./Routes";
import Navbar from "./Navbar";
//REDUX IMPORTS
import {useDispatch} from "react-redux";
import {getLanguages} from "./actionCreators/getLanguagesCreator";

function App() {
  const dispatch = useDispatch();

  useEffect(() => {

    async function getSeedData() {
      //GET AVAILABLE LANGUAGES TO TRANSLATE LYRICS TO FROM IBM API
      dispatch(getLanguages());
    };
    getSeedData();
  }, [dispatch]);

  return (
    <div className="App">
      <Navbar />
      <Routes />
    </div>
  );
};

export default App;