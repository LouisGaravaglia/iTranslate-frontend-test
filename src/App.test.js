import React from 'react'
import {render, cleanup} from '@testing-library/react'
import {Router, MemoryRouter} from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect'
import {Provider} from 'react-redux'
import {createStore} from "redux";
import {createMemoryHistory} from 'history'
import App from "./App";

afterEach(cleanup);

//CREATE A MOCK INTERSECTION OBSERVER CLASS FOR TESTING AND APPEND IT TO THE WINDOW OBJECT
const mockIntersectionObserver = class {
  constructor() {}
  observe() {}
  unobserve() {}
  disconnect() {}
};

beforeEach(() => {
  window.IntersectionObserver = mockIntersectionObserver;
});

const startingState = {
  selectedTrack: {trackName: "chanel", trackId: 42}, 
  errors: {translationError: false, languageError: false, lyricsError: false, searchError: false, generalError: false}
};

/**
 * reducer function to mimic the reducer used in Redux.
 * @param {object} state - holds an object of data used within the components
 * @param {object} action - object where the values are strings used to fire certain actions
 */
function reducer(state = startingState, action) {
  switch (action.type) {
    case "ADD_ALBUMs":
      return {...state, albums: action.albums};
    default:
      return state;
  };
};

/**
 * Function that renders the component within a redux environment since most 
 * components rely on redux for data in order to mount.
 * @param {ReactComponent} component - private spotify client id
 * @param {function} store - creates the react store using the createStore method imported from redux
 */
function renderWithRedux(
  component,
  {initialState, store = createStore(reducer, initialState)} = {}
) {
  return {
    ...render(<Provider store={store}>{component}</Provider>)
  };
};

//SMOKE TEST
describe('Smoke Test for App component', () => {

  it('renders without crashing', () => {
    const history = createMemoryHistory();

    renderWithRedux(
      <Router history={history}>
        <App />
      </Router>
    );
  });
});

//SNAPSHOT TEST
describe('Snapshot Test for App component', () => {

  it('matches snapshot', () => {
    const history = createMemoryHistory();

    const {asFragment} = renderWithRedux(
      <Router history={history}>
        <App />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Make sure appropriate text is appearing in certain routes', () => {

  it('displays the "find your song!" search bar', () => {

    const {getByText} = renderWithRedux(
      <MemoryRouter initialEntries={["/"]}>
        <App />
      </MemoryRouter>
    );
    expect(getByText("Find your song!")).toBeInTheDocument();
  });

  it('displays the "Artists Genre Danceability" in the Browse route', () => {

    const {getByText} = renderWithRedux(
      <MemoryRouter initialEntries={["/browse"]}>
        <App />
      </MemoryRouter>
    );
    expect(getByText("Artists")).toBeInTheDocument();
  });
});