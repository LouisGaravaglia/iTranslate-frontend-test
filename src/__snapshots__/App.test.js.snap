// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Navbar-Container-Left"
    >
      <div
        class="Navbar-Search-Box"
      >
        <span
          class="Hover-Box"
          style="display: inline-block; transform: scale(1);"
        >
          <a
            aria-current="page"
            class="Navbar-Search active"
            href="/"
          >
            LYRCS
          </a>
        </span>
      </div>
    </div>
    <div
      class="Navbar-Container-Right"
    >
      <div
        class="Navbar-Browse-Box"
      >
        <span
          class="Hover-Box"
          style="display: inline-block; transform: scale(1);"
        >
          <svg
            class="sc-bdVaJa fUuvxv Navbar-Browse"
            fill="#fff"
            height="35px"
            rotate="0"
            viewBox="0 0 1024 1024"
            width="35px"
          >
            <path
              d="M128 768h768v-85.332h-768v85.332zM128 554.668h768v-85.334h-768v85.334zM128 256v85.33h768v-85.33h-768z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="Flash-Messages-Container"
      />
      <div
        style="background-color: rgb(29, 77, 190);"
      >
        <div
          style="opacity: 0;"
        >
          <div
            class="Search-Field"
          >
            <div
              class="Search-Field-Content"
            >
              <h1
                class="Search-Landing"
              >
                Find your song!
              </h1>
              <form
                class="Search-Input-Container"
              >
                <div>
                  <input
                    class="Search-Landing-Text"
                    id="SearchVal"
                    name="searchVal"
                    type="text"
                    value=""
                  />
                  <button
                    type="submit"
                  >
                    <span
                      class="Hover-Box"
                      style="display: inline-block; transform: scale(1);"
                    >
                      <i
                        class="fa fa-search icon Search-Landing-Button"
                      />
                    </span>
                  </button>
                </div>
              </form>
            </div>
            <div
              class="Search-Field-Filler"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
